frontend/actions/src/main/kotlin/com/virtuslab/gitmachete/frontend/actions/dialogs/GitNewCommitMessageActionDialog.kt

```diff
@@ -0,0 +1,65 @@
+package com.virtuslab.gitmachete.frontend.actions.dialogs
+
+import com.intellij.openapi.editor.impl.EditorImpl
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.util.Disposer
+import com.intellij.openapi.vcs.ui.CommitMessage
+import com.intellij.ui.components.JBLabel
+import com.intellij.util.ui.JBUI
+import com.intellij.util.ui.UIUtil
+
+/**
+ * A custom implementation based on [git4idea.rebase.log.GitNewCommitMessageActionDialog].
+ * */
+internal class GitNewCommitMessageActionDialog(
+  val project: Project,
+  val originMessage: String,
+  title: String,
+  val dialogLabel: String
+) : DialogWrapper(project, true) {
+  val commitEditor = createCommitEditor()
+  var onOk: (String) -> Unit = {}
+
+  init {
+    Disposer.register(disposable, commitEditor)
+
+    init()
+    isModal = false
+    this.title = title
+  }
+
+  fun show(onOk: (newMessage: String) -> Unit) {
+    this.onOk = onOk
+    show()
+  }
+
+  override fun createCenterPanel() =
+    JBUI.Panels.simplePanel(UIUtil.DEFAULT_HGAP, UIUtil.DEFAULT_VGAP)
+      .addToTop(JBLabel(dialogLabel))
+      .addToCenter(commitEditor)
+
+  override fun getPreferredFocusedComponent() = commitEditor.editorField
+
+  override fun getDimensionServiceKey() = "Git.Rebase.Log.Action.NewCommitMessage.Dialog"
+
+  private fun createCommitEditor(): CommitMessage {
+    val editor = CommitMessage(project, /* withSeparator */ false, /* showToolbar */ false, /* runInspections */ true)
+    editor.text = originMessage
+    editor.editorField.setCaretPosition(0)
+    editor.editorField.addSettingsProvider { editorEx ->
+      // display at least several rows for one-line messages
+      val MIN_ROWS = 3
+      if ((editorEx as EditorImpl).visibleLineCount < MIN_ROWS) {
+        verticalStretch = 1.5F
```

Comment:
Hmmm it's some factor of how much vertical space is shown, as compared to how much the actual text takes? so what's gonna be the behavior for `visibleLineCount` being 1 or 2 lines?

Reply:
Yes. In our case we will never have less than 6 lines (2 commit messages followed by 2 new lines each). It's just to keep the window in reasonable size when the text is short. 

I ve tested the dialog for only two single-lined commit messages and dozen of them too - it works good, shows a scroll bar when needed.

Good point here, I am applying **horizontalStretch** cause the dialog title was getting trimmed when the message was short.